containerized local dev environment breaks because c++ compiler is missing

`docker build -t seagldev .` errors out with

```
...
checking for SOCK_CLOEXEC in sys/socket.h... yes
checking for sys/event.h... no
checking for epoll_create() in sys/epoll.h... yes
checking for clock_gettime()... yes
checking for CLOCK_MONOTONIC_RAW in time.h... yes
checking for CLOCK_MONOTONIC in time.h... yes
CXXFLAGS=-g -O2 -fstack-protector-strong -Wformat -Werror=format-security
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can
be found here:

  /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/eventmachine-1.2.7/mkmf.log

current directory: /var/lib/gems/2.3.0/gems/eventmachine-1.2.7/ext
make "DESTDIR=" clean

current directory: /var/lib/gems/2.3.0/gems/eventmachine-1.2.7/ext
make "DESTDIR="
compiling binder.cpp
make: g++: Command not found
Makefile:231: recipe for target 'binder.o' failed
make: *** [binder.o] Error 127

make failed, exit code 2

Gem files will remain installed in /var/lib/gems/2.3.0/gems/eventmachine-1.2.7
for inspection.
Results logged to
/var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/eventmachine-1.2.7/gem_make.out

An error occurred while installing eventmachine (1.2.7), and Bundler cannot
continue.
Make sure that `gem install eventmachine -v '1.2.7' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 191, which depends on
    jekyll-avatar was resolved to 0.6.0, which depends on
      jekyll was resolved to 3.7.3, which depends on
        em-websocket was resolved to 0.5.1, which depends on
          eventmachine
The command '/bin/sh -c bundle install' returned a non-zero code: 5
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,30 +1,15 @@
-FROM ubuntu:16.04
+FROM ruby:2.5
 
-# idea/TODO: use non-root user inside the Docker container
+RUN apt-get update && apt-get -y dist-upgrade
 
-# note: could install `build-essential` instead of individual make, gcc,
-# libghc-zlib-dev, and patch packages (but I rather like being explicit here)
+# throw errors if Gemfile has been modified since Gemfile.lock
+RUN bundle config --global frozen 1
 
-RUN apt-get update \
-    && apt-get -y dist-upgrade \
-    && apt-get install -y --no-install-recommends \
-       ruby \
-       ruby-dev \
-       make \
-       gcc \
-       libghc-zlib-dev \
-       patch \
-    && apt-get clean \
-    && apt-get -y autoremove --purge \
-    && rm -rf /var/lib/apt/lists/* \
-    && gem install bundler
+WORKDIR /usr/src/app
 
-ADD Gemfile /
-
+COPY Gemfile Gemfile.lock ./
 RUN bundle install
 
-RUN mkdir /seagl
-
-WORKDIR /seagl
+COPY . .
 
 ENTRYPOINT [ "bundle", "exec", "jekyll", "serve", "--host=0.0.0.0", "--watch" ]

--- a/README.md
+++ b/README.md
@@ -119,7 +119,7 @@ docker build -t seagldev .
 Start your dev container with
 
 ```bash
-docker run -p 4000:4000 --rm -it -v $(pwd):/seagl seagldev
+docker run -p 4000:4000 --rm -it -v "$PWD":/usr/src/app seagldev
 ```
 
 View the rendered website at <http://localhost:4000> on your host.

